client/state/countries/reducer.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { SMS, DOMAIN, PAYMENT } from './constants';
+import {
+	COUNTRIES_RECEIVE,
+	COUNTRIES_REQUEST,
+	COUNTRIES_REQUEST_FAILURE,
+	COUNTRIES_REQUEST_SUCCESS
+} from 'state/action-types';
+import { countriesSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+export const items = createReducer( { [ SMS ]: [], [ DOMAIN ]: [], [ PAYMENT ]: [] }, {
```

Comment:
Yeah that would be a lot of boilerplate. I think `COUNTRIES_RECEIVE` is fine provided we also provide the action.listType. It's fine for multiple reducers to listen to the same event.

For example:
```javascript
const smsReducer = createReducer( [], {
    [ COUNTRIES_RECEIVE ]: ( state, { listType, countries } ) => {
        if( listType !== SMS ) {
            return state;
         }
         return countries;
    } 
} );
```